File: src/SwatiFullStackApp.Api/Catalog/CatalogModels.cs
namespace SwatiFullStackApp.Api.Catalog;

public sealed record ConceptCard(
    string Title,
    string SimpleExplanation,
    string PracticalExample,
    string SeniorFocus,
    IReadOnlyList<string> KeySkills);

public sealed record CatalogSection(
    string Id,
    string Title,
    string Summary,
    IReadOnlyList<ConceptCard> Concepts);

public sealed record CatalogResponse(
    string Title,
    string Subtitle,
    IReadOnlyList<CatalogSection> Sections,
    IReadOnlyList<string> SuggestedProjects);

File: src/swati-fullstack-app-ui/src/app/models/catalog.models.ts
export interface ConceptCard {
  title: string;
  simpleExplanation: string;
  practicalExample: string;
  seniorFocus: string;
  keySkills: string[];
}

export interface CatalogSection {
  id: string;
  title: string;
  summary: string;
  concepts: ConceptCard[];
}

export interface CatalogResponse {
  title: string;
  subtitle: string;
  sections: CatalogSection[];
  suggestedProjects: string[];
}
